qubesmanager/settings.py

```diff
@@ -414,27 +438,60 @@ def _rename_vm(self, t_monitor, name):
 
     def rename_vm(self):
 
-        new_vm_name, ok = QInputDialog.getText(self, self.tr('Rename VM'), self.tr('New name: (WARNING: all other changes will be discarded)'))
+        new_vm_name, ok = QInputDialog.getText(self
+            , self.tr('Rename VM')
+            , self.tr('New name: (WARNING: '
+                      'all other changes will be discarded)'))
 
         if ok:
+            self._run_in_thread(self._rename_vm, new_vm_name)
+            self.done(0)
 
-            t_monitor = thread_monitor.ThreadMonitor()
-            thread = threading.Thread(target=self._rename_vm, args=(t_monitor, new_vm_name,))
-            thread.daemon = True
-            thread.start()
+    def _remove_vm(self, t_monitor):
+        try:
+            del self.vm.app.domains[self.vm.name]
 
-            while not t_monitor.is_finished():
-                self.qapp.processEvents()
-                time.sleep (0.1)
+        except Exception as ex:
+            t_monitor.set_error_msg(str(ex))
 
-            if not t_monitor.success:
-                QMessageBox.warning(None,
-                    self.tr("Error renaming the VM!"),
-                    self.tr("ERROR: {}").format(
-                        t_monitor.error_msg))
+        t_monitor.set_finished()
+
+    def remove_vm(self):
+
+        answer, ok = QInputDialog.getText(self, self.tr('Delete VM')
+            , self.tr('Are you absolutely sure you want to delete this VM? '
+                      '<br \> All VM settings and data will be irrevocably'
+                      ' deleted. <br \> If you are sure, please enter this '
+                      'VM\'s name below.'))
 
+
+        if ok and answer == self.vm.name:
+            self._run_in_thread(self._remove_vm)
             self.done(0)
 
+        else:
+            QMessageBox.warning(None, self.tr("Removal cancelled")
+                                , self.tr("The VM will not be removed."))
```

Comment:
Code style: keep coma at the end of previous line. And fix indentation.

Reply:
Also, not sure if canceling window by the user (`ok == False`) is worth additional message?